azkaban-exec-server/src/test/java/azkaban/common/ExecJettyServerModuleTest.java

```diff
@@ -0,0 +1,132 @@
+
+  @Test
+  public void testSslDisabledJettyServer() throws Exception {
+    ExecJettyServerModule execJettyServerModule = new ExecJettyServerModule();
+
+    Props props = new Props();
+    props.put("jetty.use.ssl", "false");
+    props.put("executor.port", JETTY_PORT);
+    Server jettyServer = execJettyServerModule.createJettyServer(props);
+    Assert.assertEquals(1, jettyServer.getConnectors().length);
+    Assert.assertTrue(jettyServer.getConnectors()[0] instanceof SocketConnector);
+    SocketConnector socketConnector = (SocketConnector) jettyServer.getConnectors()[0];
+    Assert.assertEquals(54321, socketConnector.getPort());
+    final Context root = new Context(jettyServer, "/", Context.SESSIONS);
+    root.addServlet(new ServletHolder(new SimpleServlet()), "/simple");
+    jettyServer.start();
+    final ExecutorApiClient tlsDisabledClient = new ExecutorApiClient(new Props());
+    final String postResponse = tlsDisabledClient
+        .doPost(new URI(SimpleServlet.TLS_DISABLED_URI), DispatchMethod.CONTAINERIZED,
+            null);
+    Assert.assertEquals(ExecJettyServerModuleTest.SimpleServlet.POST_RESPONSE_STRING, postResponse);
+    jettyServer.stop();
+  }
+
+  @Test
+  public void testSslEnabledJettyServer() throws Exception {
+    ExecJettyServerModule execJettyServerModule = new ExecJettyServerModule();
+
+    Props props = new Props();
+    props.put("jetty.use.ssl", "true");
+    props.put("executor.ssl.port", JETTY_TLS_PORT);
+    props.put("jetty.keystore", KEYSTORE_PATH);
+    props.put("jetty.password", DEFAULT_PASSWORD);
+    props.put("jetty.keypassword", DEFAULT_PASSWORD);
+    props.put("jetty.truststore", TRUSTSTORE_PATH);
+    props.put("jetty.trustpassword", DEFAULT_PASSWORD);
+    Server jettyServer = execJettyServerModule.createJettyServer(props);
+    Assert.assertEquals(1, jettyServer.getConnectors().length);
+    Assert.assertTrue(jettyServer.getConnectors()[0] instanceof SslSocketConnector);
+    SslSocketConnector sslSocketConnector = (SslSocketConnector) jettyServer.getConnectors()[0];
+    Assert.assertEquals(JETTY_TLS_PORT, sslSocketConnector.getPort());
+    final Context root = new Context(jettyServer, "/", Context.SESSIONS);
+    root.addServlet(new ServletHolder(new SimpleServlet()), "/simple");
+    jettyServer.start();
+
+    Props clientProps = new Props();
+    clientProps.put(EXECUTOR_CLIENT_TLS_ENABLED, "true");
+    clientProps
+        .put(EXECUTOR_CLIENT_TRUSTSTORE_PATH, TRUSTSTORE_PATH);
+    clientProps.put(EXECUTOR_CLIENT_TRUSTSTORE_PASSWORD, "changeit");
+
+    final ExecutorApiClient tlsEnabledClient = new ExecutorApiClient(clientProps);
+    final String postResponse = tlsEnabledClient
+        .doPost(new URI(SimpleServlet.TLS_ENABLED_URI), DispatchMethod.CONTAINERIZED,
+            null);
+    Assert.assertEquals(ExecJettyServerModuleTest.SimpleServlet.POST_RESPONSE_STRING, postResponse);
+    jettyServer.stop();
+  }
+
+  private static class SimpleServlet extends HttpServlet {
```

Comment:
Maybe be make this class public in `ExecutorApiClient` and use it directly ?

Reply:
Done.